cpp/src/binaryop/jit/code/operation.cpp

```diff
@@ -400,6 +400,41 @@ const char* operation =
             return LogBase::operate<TypeOut, TypeLhs,TypeRhs>(y, x);
         }
     };
+
+    struct ATan2 {
+        // Ideally, these two specializations - one for integral types and one for non integral
+        // types shouldn't be required, as std::atan2 should promote integral types automatically
+        // to double and call the std::atan2 overload for doubles. Sadly, doing this in jitified
+        // code does not work - it is having trouble deciding between float/double overloads. Hence,
+        // promote these types manually to double, for what should have happened naturally
+        template <typename TypeOut,
```

Comment:
Hmm.  It might be worth holding off on this PR until this other one goes in:

https://github.com/rapidsai/cudf/pull/4854

It has a general fix for dealing with type conversions to handle loss of precision.

Reply:
@nvdbaranec even with #4854 this change is required to deal with `atan2` as #4854 is only using common type for *existing* operators.

further, i'm not sure if uniformly applying an ulp of 4 is future proof, as there are operations that yield more than this (in case we decide to support them later). please see [here](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#standard-functions)